connector_file_bank_statement/unit/load_policy.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Georges Racinet
+#    Copyright 2014 Anybox SAS
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import csv
+import osv
+import simplejson as json
+from binascii import b2a_base64
+from cStringIO import StringIO
+from datetime import datetime
+
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+from openerp.addons.connector_file.unit.move_load_policy import MoveLoadPolicy
+
+from ..backend import file_import_bank_statement
+
+
+def now_str():
+    return datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+# here it'd be more logical if the base class would be called simply
+# LoadPolicy (and its parent LoadPolicy should be called BaseLoadPolicy)
+@file_import_bank_statement
+class StatementLoadPolicy(MoveLoadPolicy):
+    """Defines the load process of CSV chunks as bank statements."""
+
+    def load_one_chunk(self, chunk_b_id):
+        """Load chunk as a bank statement.
+
+        Notes:
+
+        * the chunk content will end up as attachment on the statement object
+          as well (poor DB)
+        * for now I'm forced to reconstruct the CSV, to give it to statement
+          profile parser (base64 and all). Inefficient and another source of
+          hardcoding (dialect). It'd be so much better to use an entry point
+          meant for after parsing.
+        * auto-completion is already taken care of by import profile, but that
+          means the configuration must be appropriate. TODO
+          can it be correct to confirm a statement that has not been
+          auto-completed ? If not, the method should raise an error if the
+          profile is not an auto-completing one.
+        """
+        s = self.session
+        profile_obj = s.pool['account.statement.profile']
+
+        chunk_b = s.pool[self._model_name].browse(s.cr, s.uid, chunk_b_id,
+                                                  context=s.context)
+        backend_b = chunk_b.backend_id
+        if chunk_b.load_state != 'pending':
```

Comment:
Shouldn't this be only on 'done'? If your import failed for a bad configuration of backend like wrong user, you cannot restart it.